### PR TITLE
fix: prevent scrolling while widget not expanded

### PR DESCRIPTION
**Pull request by dennyscode**
Ticket: https://lifi.atlassian.net/browse/LF-9007

### CODE REVIEW

File: src/components/FeatureCards/FeatureCards.tsx
Comment (dennyscode) [inline src/components/FeatureCards/FeatureCards.tsx:1]: This file contains the widget-expanded listener which has been moved to it´s own hook.

File: src/components/Widgets/Widget.style.ts [hunk 1]
```diff
@@ -19,9 +21,15 @@ export const WidgetWrapper = styled(Box, {
...
-  // },
+
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    // height: 200,
+    ...(!isExpanded && {
+      '& + div': {
+        height: 496,
+      },
+    }),
+  },
```
Comment (dennyscode) [inline src/components/Widgets/Widget.style.ts:32]: While widget is not expanded, overwrite height.

File: src/components/Widgets/Widgets.style.tsx [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const WidgetContainer = styled(Box, {
       },
 
       [theme.breakpoints.up('sm' as Breakpoint)]: {
+        height: 496,
```
Comment (dennyscode) [inline src/components/Widgets/Widgets.style.tsx:42]: Height of the default widget while not expanded.

File: src/components/Widgets/Widget.style.ts [hunk 3]
```diff
@@ -19,9 +21,14 @@ export const WidgetWrapper = styled(Box, {
...
   }),
   zIndex: 2,
-  // [theme.breakpoints.up('sm' as Breakpoint)]: {
-  //   maxWidth: 416,
-  // },
+
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    ...(!isExpanded && {
+      '& > div': {
+        height: 616,
```
Comment (dennyscode) [inline src/components/Widgets/Widget.style.ts:28]: This is the minimal widget height we should enable - without Wallet-Address field we can go as low as 496px:
image

File: src/components/Widgets/Widget.tsx [hunk 4]
```diff
@@ -180,39 +179,34 @@ export function Widget({
...
     multisigSdkConfig,
-    multisigWidget,
-    setWalletSelectMenuState,
-    starterVariant,
-    theme.breakpoints,
-    theme.palette.accent1.main,
-    theme.palette.grey,
-    theme.palette.mode,
-    theme.palette.surface1.main,
-    theme.palette.surface2.main,
```
Comment (tcheee) [inline src/components/Widgets/Widget.tsx:201]: @dennyscode why so many changes inside this file?

File: src/components/Widgets/Widget.style.ts [hunk 5]
```diff
@@ -19,9 +21,14 @@ export const WidgetWrapper = styled(Box, {
     },
   }),
   zIndex: 2,
-  // [theme.breakpoints.up('sm' as Breakpoint)]: {
-  //   maxWidth: 416,
-  // },
+
+  [theme.breakpoints.up('sm' as Breakpoint)]: {
+    ...(!isExpanded && {
```
Comment (admsk2) [inline src/components/Widgets/Widget.style.ts:26]: i think it is quite difficult to read that part, maybe that could be refactored like:

```
export const WidgetWrapper = styled(Box, {
  shouldForwardProp: (prop) => !['welcomeScreenClosed', 'isExpanded'].includes(prop),
})<WidgetWrapperProps>(({ theme, welcomeScreenClosed, isExpanded }) => {
  const baseStyle = {
    width: '100%',
    position: 'relative',
    margin: theme.spacing(0, 'auto'),
    zIndex: 2,
  };

  const hoverEffect = !welcomeScreenClosed ? {
    '&:hover': {
      marginTop: 0,
    }
  } : {};

  const responsiveHeight = theme.breakpoints.up('sm') as Breakpoint;
  const expandedStyle = !isExpanded ? {
    '& > div': {
      height: 616,
    }
  } : {};

  return {
    ...baseStyle,
    ...hoverEffect,
    [responsiveHeight]: {
      ...expandedStyle
    }
  };
});
```

File: src/components/Widgets/Widget.tsx (hunk 4 above)
Comment (dennyscode) [reply src/components/Widgets/Widget.tsx:201]: For that, I just updated the dependency array and it cleaned up some.

File: src/components/Widgets/Widget.style.ts (hunk 5 above)
Comment (dennyscode) [reply src/components/Widgets/Widget.style.ts:26]: Thanks for the input @admsk2 !

Maybe we can have this discussion with @oktapodia and tcheee. 
Personally, I do not find the approach you suggested more readable but we should decide within our team what we see as best practise for such "more sophisticated" styled-components. Wdyt?

**Review by admsk2 — APPROVED** [review]

**Review by chybisov — CHANGES_REQUESTED** [review]
Found this while testing

https://github.com/user-attachments/assets/b55d721d-0218-4e19-b203-9063d8ca81de

**Comment by dennyscode** [comment]
> Found this while testing
> 
>  Screen.Recording.2024-08-20.at.12.49.56.PM.mov

[bug-not-reproducable-screen-capture.webm](https://github.com/user-attachments/assets/d342ba2a-d910-4728-b75b-9936251d135e)

Thanks Eugene! I believe this issue must have occured before we got to fix and merge into develop. I cannot reproduce it myself. Let me know if I am wrong and we still have the "flickering" or "jumping" of the widget.

**Pull request closed without merging**